fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/cascades/values/simplification/ValueComputationTest.java

```diff
@@ -117,6 +120,60 @@ void testPullUpFieldValueThroughStreamingAggregation() {
         Assertions.assertEquals(expectedResult, resultsMap);
     }
 
+    @Test
+    void testPullUpValue1() {
+        final var someCurrentValue = QuantifiedObjectValue.of(ALIAS, someRecordType());
+        // _
+        final var pulledThroughValue =
+                QuantifiedObjectValue.of(ALIAS, someRecordType());
+
+        // recordType(alias)
+        final var record_type = new RecordTypeValue(QuantifiedObjectValue.of(ALIAS, new Type.AnyRecord(true)));
+        final var _a_b = FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("a", "ab"));
+        final var _x_b = FieldValue.ofFieldNames(someCurrentValue, ImmutableList.of("x", "xb"));
+        // _.a.ab + _.x.xb
+        final var _a_ab__plus__x_xb = (Value)new ArithmeticValue.AddFn().encapsulate(ImmutableList.of(_a_b, _x_b));
+        final var toBePulledUpValues = ImmutableList.of(record_type, _a_ab__plus__x_xb);
+
+        final var resultsMap = pulledThroughValue.pullUp(toBePulledUpValues, AliasMap.emptyMap(), ImmutableSet.of(), UPPER_ALIAS);
+        final var upperCurrentValue = QuantifiedObjectValue.of(UPPER_ALIAS, pulledThroughValue.getResultType());
+
+        final var new_a_b = FieldValue.ofFieldNames(upperCurrentValue, ImmutableList.of("a", "ab"));
+        final var new_x_b = FieldValue.ofFieldNames(upperCurrentValue, ImmutableList.of("x", "xb"));
+
+        final var expectedMap = ImmutableMap.of(
+                record_type, new RecordTypeValue(QuantifiedObjectValue.of(UPPER_ALIAS, new Type.AnyRecord(true))),
+                _a_ab__plus__x_xb, (Value)new ArithmeticValue.AddFn().encapsulate(ImmutableList.of(new_a_b, new_x_b)));
+        Assertions.assertEquals(expectedMap, resultsMap);
+    }
+
+    @Test
+    void testPullUpValue2() {
```

Comment:
Was one of these tests supposed to look at `recordType(FieldValue(blah))`? I'm pretty sure that that kind of transformation works, but I don't see a test for it.